pkg/alertmanager/api_grafana.go

```diff
@@ -309,8 +309,25 @@ func (am *MultitenantAlertmanager) SetUserGrafanaConfig(w http.ResponseWriter, r
 		return
 	}
 
-	payload, err := io.ReadAll(r.Body)
+	var input io.Reader
+	maxConfigSize := am.limits.AlertmanagerMaxGrafanaConfigSize(userID)
+	if maxConfigSize > 0 {
+		input = http.MaxBytesReader(w, r.Body, int64(maxConfigSize))
+	} else {
+		input = r.Body
+	}
+
+	payload, err := io.ReadAll(input)
 	if err != nil {
+		maxBytesErr := &http.MaxBytesError{}
```

Comment:
No need for this to be a pointer (but see comment below, I don't think we need to keep it)

```suggestion
		var maxBytesErr http.MaxBytesError
```

Reply:
It does need to be a pointer actually. `errors.As` takes a pointer to the target which implements `error`. Since `http.MaxBytesError` has a pointer receiver on `Error`, the double indirection is required I've learned this today TBH 😅 
![Screenshot 2024-09-25 at 13 57 35](https://github.com/user-attachments/assets/84a2e650-e869-4f86-b842-94a9d67a42d5)

Reply:
Wow, TIL. That's.... odd.